From 81f8ad58ed080b80674058ea9c6a40222f2990c2 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Sat, 12 Sep 2026 15:53:08 -0400
Subject: [PATCH] nuke legacy setup.py and replace with pyproject.toml

Convertion with claude on local Qwen3-Coder-Next@Q4 + human edit.
---
 pyproject.toml | 47 ++++++++++++++++++++++++++++++
 setup.py       | 77 --------------------------------------------------
 2 files changed, 47 insertions(+), 77 deletions(-)
 create mode 100644 pyproject.toml
 delete mode 100644 setup.py

diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000..347a07f
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,47 @@
+[build-system]
+requires = ["setuptools>=61", "setuptools_scm>=8.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "logfury"
+description = "Toolkit for responsible, low-boilerplate logging of library method calls"
+readme = "README.rst"
+license = {"text" = "BSD"}
+keywords = ["logging", "tracing"]
+classifiers = [
+    "Development Status :: 6 - Mature",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Topic :: System :: Logging",
+    "Natural Language :: English",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.5",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+]
+authors = [
+    {name = "Pawel Polewicz", email = "p.polewicz@gmail.com"}
+]
+dynamic = ["version"]
+
+[project.urls]
+Homepage = "https://github.com/reef-technologies/logfury"
+"Download URL" = "https://github.com/reef-technologies/logfury/releases"
+
+[tool.setuptools]
+packages = ["logfury"]
+
+[tool.setuptools.package-data]
+logfury = ["requirements.txt", "LICENSE"]
+
+[tool.setuptools_scm]
+version_scheme = "no-guess-dev"
+local_scheme = "no-local-version"
diff --git a/setup.py b/setup.py
deleted file mode 100644
index b244da4..0000000
--- a/setup.py
+++ /dev/null
@@ -1,77 +0,0 @@
-# noqa
-
-import os.path
-from codecs import open
-
-from setuptools import find_packages, setup
-
-################################################################### yapf: disable
-
-NAME         = 'logfury'
-
-AUTHOR       = 'Pawel Polewicz'
-AUTHOR_EMAIL = 'p.polewicz@gmail.com'
-
-DESCRIPTION  = 'Toolkit for responsible, low-boilerplate logging of library method calls',
-LICENSE      = 'BSD'
-KEYWORDS     = ['logging', 'tracing']
-URL          = 'https://github.com/reef-technologies/logfury'
-DOWNLOAD_URL = URL + '/releases'
-
-CLASSIFIERS = [
-    'Development Status :: 6 - Mature',
-
-    'Intended Audience :: Developers',
-    'Topic :: Software Development :: Libraries :: Python Modules',
-    'Topic :: System :: Logging',
-
-    'Natural Language :: English',
-
-    'License :: OSI Approved :: BSD License',
-
-    'Operating System :: OS Independent',
-    'Programming Language :: Python',
-    'Programming Language :: Python :: 3',
-    'Programming Language :: Python :: 3.5',
-    'Programming Language :: Python :: 3.6',
-    'Programming Language :: Python :: 3.7',
-    'Programming Language :: Python :: 3.8',
-    'Programming Language :: Python :: 3.9',
-    'Programming Language :: Python :: 3.10',
-    'Programming Language :: Python :: Implementation :: CPython',
-    'Programming Language :: Python :: Implementation :: PyPy',
-]
-
-################################################################### yapf: enable
-
-here = os.path.abspath(os.path.dirname(__file__))
-
-
-def read_file_contents(filename):
-    with open(os.path.join(here, filename), 'rb', encoding='utf-8') as f:
-        return f.read()
-
-
-setup(
-    name             = NAME,
-    url              = URL,
-    download_url     = DOWNLOAD_URL,
-
-    author           = AUTHOR,
-    author_email     = AUTHOR_EMAIL,
-    maintainer       = AUTHOR,
-    maintainer_email = AUTHOR_EMAIL,
-
-    packages         = find_packages(exclude=['test*']),
-    license          = LICENSE,
-
-    description      = DESCRIPTION,
-    long_description = read_file_contents('README.rst'),
-    keywords         = KEYWORDS,
-
-    classifiers      = CLASSIFIERS,
-    package_data     = {NAME: ['requirements.txt', 'LICENSE']},
-
-    setup_requires   = ['setuptools_scm<6.0'],  # setuptools_scm>=6.0 doesn't support Python 3.5
-    use_scm_version  = True,
-)  # yapf: disable
-- 
2.55.0

